src/conda_devenv/devenv.py

```diff
@@ -27,13 +28,120 @@
 YAMLData = dict[str, Any]
 
 
+@dataclass(frozen=True)
+class ProcessedEnvironment:
+    """
+    Result of processing a .devenv.yml or a plain .yml file.
+    """
+
+    conda_yaml_dict: YAMLData
+    rendered_yaml: str
+    is_devenv_file: bool
+
+    @classmethod
+    def from_file(cls, p: Path, *, conda_platform: CondaPlatform | None = None) -> Self:
+        is_devenv_file = p.suffixes == [".devenv", ".yml"]
+
+        conda_yaml_dict = load_yaml_dict(p, conda_platform=conda_platform)
+        if is_devenv_file:
+            rendered_yaml = render_for_conda_env(conda_yaml_dict)
+        else:
+            rendered_yaml = p.read_text(encoding="UTF-8")
+        return cls(
+            conda_yaml_dict=conda_yaml_dict,
+            rendered_yaml=rendered_yaml,
+            is_devenv_file=is_devenv_file,
+        )
+
+
+class CondaPlatform(Enum):
```

Comment:
Since this is going to probably be a new major version, maybe we could drop support for 32-bit platforms and add proper support for:

* `linux-aarch64`
* `linux-ppc64le`
* `osx-arm64`

?

I think they still follow the conda convention, the `32` or `64` is replaced by `aarch64` or `arm64` for example. But this could be a separate PR later.

Reply:
Yeah I think a separate PR would be good.